core/src/main/java/io/questdb/network/IODispatcherOsx.java

```diff
@@ -52,17 +55,40 @@ public void close() {
         LOG.info().$("closed").$();
     }
 
+    private static boolean isEventId(long id) {
+        return (id & 1) == 1;
+    }
+
+    private void doDisconnect(C context, long id, int reason) {
+        final SuspendEvent suspendEvent = context.getSuspendEvent();
+        if (suspendEvent != null) {
+            // yes, we can do a binary search over EVM_OPERATION_ID since
+            // these ref ids are monotonically growing
+            int eventRow = pendingEvents.binarySearch(id, EVM_OPERATION_ID);
+            if (eventRow < 0) {
+                LOG.error().$("internal error: suspend event not found [id=").$(id).I$();
```

Comment:
this should be critical

Reply:
Changed in bbec37baf832db9a846d04e59ab124f8bb84eadc